gnucash/register/register-gnome/completioncell-gnome.c

```diff
@@ -0,0 +1,1019 @@
+
+void
+gnc_completion_cell_set_value (CompletionCell* cell, const char* str)
+{
+    if (!cell || !str)
+
+    gnc_basic_cell_set_value (&cell->cell, str);
+}
+
+static inline void
+list_store_append (GtkListStore *store, char* string,
+                   char* markup, gint weight)
+{
+    GtkTreeIter iter;
+
+    g_return_if_fail (store);
+    g_return_if_fail (string);
+    g_return_if_fail (markup);
+
+    gtk_list_store_append (store, &iter);
+
+    gtk_list_store_set (store, &iter, TEXT_COL, string,
+                                      TEXT_MARKUP_COL, markup,
+                                      WEIGHT_COL, weight, -1);
+}
+
+static char*
+normalize_and_fold (char* utf8_string)
+{
+    char *normalized = g_utf8_normalize (utf8_string, -1, G_NORMALIZE_NFC);
+    if (!normalized)
+        return NULL;
+
+    char *folded = g_utf8_casefold (normalized, -1);
+    g_free (normalized);
+    return folded;
+}
+
+static void
+add_item (gpointer key, gpointer value, gpointer user_data)
+{
+    PopBox* box = user_data;
+    gchar *hash_entry = g_strdup (key);
+
+    if (hash_entry && *hash_entry)
+    {
+        gnc_utf8_strip_invalid_and_controls (hash_entry);
+        gchar *key_norm_fold = normalize_and_fold (hash_entry);
+        gchar *found = g_strstr_len (key_norm_fold, -1, box->newval);
+
+        if (found)
+        {
+            gchar *markup = NULL, *prefix = NULL, *match = NULL, *suffix = NULL;
+            gint prefix_len;
+            gint occurrence_difference;
+            gint weight;
+            gboolean have_boundary = FALSE;
+            glong len = g_utf8_strlen (box->newval, -1);
+            gulong loc = g_utf8_pointer_to_offset (key_norm_fold, found);
+            OccurrenceData *od = (OccurrenceData*)value;
+
+            if (box->register_is_reversed)
+                occurrence_difference = od->occurrence + 1;
+            else
+                occurrence_difference = box->occurrence - od->occurrence;
+
+            if (loc > 0)
+                prefix = g_utf8_substring (hash_entry, 0, loc);
+            else
+                prefix = g_strdup ("");
+            prefix_len = g_utf8_strlen (prefix, -1);
+
+            match = g_utf8_substring (hash_entry, loc, loc + len);
+
+            if (loc > 1)
+            {
+                gunichar prev = g_utf8_get_char (g_utf8_offset_to_pointer (hash_entry, loc - 1));
+                if (prev && (g_unichar_isspace (prev) || g_unichar_ispunct (prev)))
+                    have_boundary = TRUE;
+            }
+
+            suffix = g_utf8_substring (hash_entry, loc + len, g_utf8_strlen (hash_entry, -1));
+
+            if (g_utf8_strlen (suffix, -1) == 0)
+                markup = g_markup_printf_escaped ("%s<b>%s</b>%s", prefix, match, " ");
+            else
+                markup = g_markup_printf_escaped ("%s<b>%s</b>%s%s", prefix, match, suffix, " ");
+
+            if (((prefix_len != 0 ) && !have_boundary) || (box->time_now - od->date > 157680000)) // aprox 5 years
```

Comment:
Perhaps more clear to put the number above also as a constant that shows the multiplication you did to get to approximately 5 years, like:
```
const time64 five_years_secs = 5 * 365 * 24 * 60 * 60;
```

Reply:
Having said that, strongly penalizing descriptions that are older than a fixed time interval compared to ones that are more recent than this time interval seems arbitrary. We already sort on occurrence_interval so older entries already sink down in the list. Did you do experiments that suggested this is useful ?

Reply:
> Perhaps more clear to put the number above also as a constant that shows the multiplication you did to get to approximately 5 years, like:
> 
> ```
> const time64 five_years_secs = 5 * 365 * 24 * 60 * 60;
> ```
Can do.

Reply:
> Having said that, strongly penalizing descriptions that are older than a fixed time interval compared to ones that are more recent than this time interval seems arbitrary. We already sort on occurrence_interval so older entries already sink down in the list. Did you do experiments that suggested this is useful ?

I think this was mentioned above by @dawansv that he did not want to see old entries that other wise would of fallen into the most recent in time top section. In the example above, keying 'b' would add the old transaction 'Bank Account' in the top section, instead it is in the appropriate place in time in the lower section..

Reply:
I don't recall specifically what I said (and I am no stranger to contradicting myself even in the same breath). Also worth noting that the last version I tested and liked does include that code. That said, intuitively I agree with @gjanssens here and @jralls further below that in theory we shouldn't need an arbitrary time limit. Older entries should already be pushed down a bit by the "recent in time" preference then move up progressively as we type more, and more recent matches start going away. In truth until I can actually try it it's hard to tell with 100% certainty whether this is needed or not. I think from a user experience perspective this feature has a lot to do with how it feels even if it's not entirely logical, hence why it's a bit harder to tell without testing.